Bug: test:types broken with eslint>=9.31.0

### Environment

Node version: 24.4.1
npm version: 11.4.2
ESLint version: 9.31.0
eslint-plugin-n version: 17.21.0
Operating System: macOS


### What rule do you want to report?

The CI process - test:types

### Link to Minimal Reproducible Example

npm run test:types

### What did you expect to happen?

The CI should pass

### Participation

- [x] I am willing to submit a pull request for this issue.

### Additional comments

Since eslint@9.31.0 was released, the CI fails:
```
➜  eslint-plugin-n git:(master) npm run test:types
npm warn using --force Recommended protections disabled.

> eslint-plugin-n@17.21.0 test:types
> tsc --noEmit --emitDeclarationOnly false

lib/rules/no-path-concat.js:201:67 - error TS2339: Property 'getScope' does not exist on type 'PluginRuleContext<{ RuleOptions: unknown[]; }>'.

201                     sourceCode.getScope?.(programNode) ?? context.getScope()
                                                                      ~~~~~~~~

lib/util/eslint-compat.js:20:69 - error TS2339: Property 'getScope' does not exist on type 'RuleContext'.

20     return sourceCode.getScope?.(node || sourceCode.ast) || context.getScope()
                                                                       ~~~~~~~~

lib/util/eslint-compat.js:28:55 - error TS2339: Property 'getAncestors' does not exist on type 'RuleContext'.

28     return sourceCode.getAncestors?.(node) || context.getAncestors()
                                                         ~~~~~~~~~~~~


Found 3 errors in 2 files.

Errors  Files
     1  lib/rules/no-path-concat.js:201
     2  lib/util/eslint-compat.js:20
```

--- a/lib/rules/no-path-concat.js
+++ b/lib/rules/no-path-concat.js
@@ -198,6 +198,7 @@ module.exports = {
             "Program:exit"(programNode) {
                 const sourceCode = getSourceCode(context)
                 const globalScope =
+                    // @ts-ignore
                     sourceCode.getScope?.(programNode) ?? context.getScope()
                 const tracker = new ReferenceTracker(globalScope)
                 /** @type {Set<import('estree').Node>} */

--- a/lib/util/eslint-compat.js
+++ b/lib/util/eslint-compat.js
@@ -17,6 +17,7 @@ exports.getScope = function (
     /** @type {Node} */ node
 ) {
     const sourceCode = exports.getSourceCode(context)
+    // @ts-ignore
     return sourceCode.getScope?.(node || sourceCode.ast) || context.getScope()
 }
 
@@ -25,6 +26,7 @@ exports.getAncestors = function (
     /** @type {Node} */ node
 ) {
     const sourceCode = exports.getSourceCode(context)
+    // @ts-ignore
     return sourceCode.getAncestors?.(node) || context.getAncestors()
 }
 

--- a/package.json
+++ b/package.json
@@ -34,7 +34,7 @@
         "@types/node": "^20.17.5",
         "@typescript-eslint/parser": "^8.26.1",
         "@typescript-eslint/typescript-estree": "^8.26.1",
-        "eslint": "^9.14.0",
+        "eslint": "^9.31.0",
         "eslint-config-prettier": "^9.1.0",
         "eslint-doc-generator": "^1.7.1",
         "eslint-plugin-eslint-plugin": "^6.3.1",

